articles/customizing-django-admin/index.md

```diff
@@ -0,0 +1,245 @@
+
+```bash
+$ .venv\Scripts\activate.bat
+```
+
+Now we can install Django in the virtual environment.
+
+```bash
+$ pip install django
+$ pip install django
+```
+###  Creating the Models
+Add the following code to your *models.py* to create some models.
+
+```py
+from django.db.models.deletion import CASCADE
+
+class  Category(models.Model):
+  cat_romance  =  "Romance"
+  cat_fantacy  =  "Fantacy"
+  cat_thriller=  "Thriller"
+  cat_horror  =  "Horror"
+  cat_crime  =  "Crime"
+  cat_true_story=  "True Story"
+  category  =  models.CharField(
+      max_length=100,
+      choices=(
+      (cat_crime, cat_crime),
+      (cat_fantacy, cat_fantacy),
+      (cat_horror, cat_horror),
+      (cat_romance, cat_romance),
+      (cat_thriller, cat_thriller),
+      (cat_true_story, cat_true_story)
+        )
+        )
+
+  def  __str__(self):
+      return  self.category
+
+class  Publisher(models.Model):
+  publisher_name  =  models.CharField(max_length=100)
+  publish_date  =  models.DateField
+
+  def  __str__(self):
+      return  self.publisher_name
+
+class  Author(models.Model):
+  gender_male  =  "Male"
+  gender_female  =  "Female"
+  gender_other  =  "Other"
+  name  =  models.CharField(max_length=100)
+  gender  =  models.CharField(max_length=100,
+      choices=(
+      (gender_female, gender_female),
+      (gender_male, gender_male),
+      (gender_other, gender_other)
+      )
+      )
+  country  =  models.CharField(max_length=100)
+
+  def  __str__(self):
+      return  self.name
+
+class  Details(models.Model):
+  book_name  =  models.CharField(max_length=100)
+  category = models.ForeignKey(Category, on_delete=CASCADE)
+  pages = models.IntegerField(default=1)
+  publisher  =  models.ForeignKey(Publisher, on_delete=models.CASCADE)
+  Author  =  models.ForeignKey(Author, on_delete=CASCADE)
+
+  def  __str__(self):
+      return  self.book_name
+```
+We can now go ahead and register our models in *admin.py* by adding the following lines of code.
+```py
+from django.contrib import admin
+from django.db import models
+from .models import Category, Publisher ,Details, Author
+
+class  categoryAdmin(admin.ModelAdmin):
+  pass
+
+class  publisherAdmin(admin.ModelAdmin):
+  pass
+
+class  detailsAdmin(admin.ModelAdmin):
+  pass
+
+class  authorAdmin(admin.ModelAdmin):
+  pass
+
+admin.site.register(Category, categoryAdmin)
+admin.site.register(Publisher, publisherAdmin)
+admin.site.register(Details, detailsAdmin)
+admin.site.register(Author, authorAdmin)  
+```
+Then make the migrations and migrate using the below commands.
+
+```bash
+$ py manage.py makemigrations
+$ py manage.py migrate
+```
+Now, let us create a superuser in order to start customizing our admin site.
+```bash
+$ py manage.py createsuperuser
+```
+Now run the development server using the command below:
+```bash
+$ py manage.py runserver
+```
+You can now log in using the superuser credentials at http://127.0.0.1:8000/admin/. You should see something similar to this.
```

Comment:
Looks like there is a missing image here. Please add the image

Reply:
not really but it's now fixed